compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -21,6 +23,14 @@ struct InteriorVisitor<'a, 'tcx> {
     expr_count: usize,
     kind: hir::GeneratorKind,
     prev_unresolved_span: Option<Span>,
+    /// Match arm guards have temporary borrows from the pattern bindings.
+    /// In case there is a yield point in a guard with a reference to such bindings,
+    /// such borrows can span across this yield point.
+    /// As such, we need to track these borrows and record them despite of the fact
+    /// that they may succeed the said yield point in the post-order.
+    nested_scope_of_guards: SmallVec<[SmallVec<[HirId; 4]>; 4]>,
```

Comment:
The outer SmallVec could probably be of len 1, since nesting these would be extremely rare in practice.